adal/src/main/java/com/microsoft/aad/adal/StorageHelper.java

```diff
@@ -304,22 +304,30 @@ public String decrypt(final String encryptedBlob)
      */
     synchronized SecretKey loadSecretKeyForEncryption() throws IOException,
             GeneralSecurityException {
+        final byte[] secretKeyData = AuthenticationSettings.INSTANCE.getSecretKeyData();
+        return loadSecretKeyForEncryption(secretKeyData == null ? VERSION_ANDROID_KEY_STORE : VERSION_USER_DEFINED);
+    }
+
+    /**
+     * Get Secret Key based on API level to use in encryption. Decryption key
+     * depends on version# since user can migrate to new Android.OS
+     *
+     * @param defaultBlobVersion the blobVersion to use by default
+     * @return SecretKey Get Secret Key based on API level to use in encryption.
+     * @throws GeneralSecurityException
+     * @throws IOException
+     */
+    synchronized SecretKey loadSecretKeyForEncryption(String defaultBlobVersion) throws IOException,
```

Comment:
Which is to say: I understand it cannot be both `private` and effectively unit tested. My concern is that it shares name with another method and if our intention is not to invoke this function from another file this may be making a hole that's easy to fall into.